src/main/webapp/scripts/script.js

```diff
@@ -0,0 +1,132 @@
+      document.getElementById('login-status').textContent = 'Signed out.';
+    }
+    document.getElementById('login').disabled = false;
+  });
+}
+
+/** 
+ * If user is currently logged in, log them out; if logged out, log them in.
+ */
+function toggleLogin() {
+  // If the user is signed in, sign them out.
+  if (firebase.auth().currentUser) {
+    firebase.auth().signOut();
+  } else {  // Sign in.
+    var email = document.getElementById('email').value;
+    var password = document.getElementById('password').value;
+
+    if (email == null || email.length < 2) {
+      alert('Please enter an email address.');
+      return;
+    }
+    if (password == null || password.length < 2) {
+      alert('Please enter a password.');
+      return;
+    }
+
+    firebase.auth()
+        .signInWithEmailAndPassword(email, password)
+        .then(function onSuccess() {
+          // TODO: send token to server
+            // firebase.auth()
+            //   .currentUser.getIdToken(/* forceRefresh */ true)
+            //   .then(function(idToken) {
+            //     // Body type defaults to JSON.
+            //     fetch('/login', {method: 'POST', body: idToken});
+            //   })
+            //   .catch(function onFailure(error) {
+            //     alert('Token retrieval failed: ' + error);
+            //   }); 
+        })
+        .catch(function onFailure(error) {
+          var errorCode = error.code;
+          var errorMessage = error.message;
+
+          if (errorCode === 'auth/wrong-passcode') {
+            alert('Wrong password.');
+          } else {
+            alert(errorMessage);
+          }
+        });
+    // document.getElementById("login").disabled = true;
+  }
+}
+
+/**
+ * Attempt to create a new user account using the email/password HTML fields.
+ */
+function onSignup() {
+  var email = document.getElementById("email").value;
+  var password = document.getElementById("password").value;
+
+  if (email == null || email.length < 2) {
+    alert("Please enter a valid email address.");
+  }
+  if (password == null || password.length < 2) {
+    alert("Please enter a valid password.");
+  }
+  
+  // If email/pass are of appropriate length, try to create a new account.
+  firebase.auth().createUserWithEmailAndPassword(email, password)
+      .catch(function onFailure(error) {
+        var errorCode = error.code;
+        var errorMessage = error.message;
+
+        if (errorCode == 'auth/weak-password') {
+          alert('The password is too weak.');
+        } else {
+          alert(errorMessage);
+        }
+  });
+}
+
+function sendPasswordResetEmail() {
+  var email = document.getElementById("email").value;
+    
+  firebase.auth().sendPasswordResetEmail(email).then(function onSuccess() {
+    alert('Password Reset Email Sent!');
+    }).catch(function(error) {
+      // Handle errors.
+      var errorCode = error.code;
+      var errorMessage = error.message;
+    
+      if (errorCode == 'auth/invalid-email') {
```

Comment:
You can just OR these two conditions together. What happens with other errors?

Reply:
Thanks for the catch here, this is really badly written code on my part.